### ISSUE TITLE
Bug with the watch-dir option in the config file

### ISSUE THREAD

**Issue by flaviomoringa** [xaionaro/clsync#148]
I'm having a strange bug (maybe a mis-configuration on my part...)

I have this config oon the config file:
[obb-mirror21-1t]
label               = obb-mirror21-1t
watch-dir           = /obb/1
destination-dir     = rsync://xxx.xxx.xxx.xxx/obb/1/
delay-sync          = 5
delay-collect       = 1
full-initialsync    = 1
mode                = rsyncdirect
retries             = 0
timeout-sync        = 3600
ignore-exitcode     = 23,24
-- = --delete -avzK %watch-dir%/ %destination-dir%/

This doesn't work:
clsync -K obb-mirror21-1t
Error: Got error while realpath() on "/obb/1 " [#0]. (2: No such file or directory)
Error: Cannot lstat64() on "/obb/1 " (2: No such file or directory)
Error: Got error while chdir("/obb/1 ") (2: No such file or directory)

This works:
clsync -K obb-mirror21-1t -W /obb/1
sending incremental file list
deleting 0/.main-4066-com-wb-wwe-brawler2014-101255a71b991eb5631ae57470792dee.obb.RnGGIs
0/
0/main-4066-com-wb-wwe-brawler2014-101255a71b991eb5631ae57470792dee.obb

Why is the watch-dir giving an error when on ly on the config file?

Thanks for any help...

**Comment by xaionaro** [comment]
Hello. Sorry for delay.

I cannot repeat the bug. Questions:
# Which version do you use (the full output of "clsync --version")?
# Can you try "--watch-dir" instead of "-W" in the command line?
# Can you run with "-b0 -d9 -Ystderr" and show the output?

**Comment by boussou** [comment]
Hi, same issue here: it seems that when you wan to use a config block, you need to pass 2 args 
example for a block named _subs_ in the conf file:

```
$ clsync -K subs
Error: argument "1 " can't be parsed as a number
```
but this works:
```
$ clsync -d1 -K subs
Debug1 (pid: 26734; thread: 0x7f15225b2780): configs_parse(): Trying config-file "/home/nadir/.clsync.conf" (case #1)
...etc
```

**Comment by xaionaro** [comment]
@boussou thank you for the report. I'll fix it today.

**Comment by boussou** [comment]
This application is a masterpiece. I am 46, I can see that ;-)
Bravo !

**Comment by xaionaro** [comment]
@boussou thank you :)

I've added the trimming. Also now it reports about where was the problem:
```
$ ./clsync -b'a0 '
Error: argument "a0" can't be parsed as a number
Error: Unable to process option "background" (with argument: "a0") from "cli_arguments"
```
(it prints the option name and the source of the option [like `cli_arguments`, `configuration`, `defaults`], now)

If everything OK then I can bump the version.

@flaviomoringa it seems you had a space after `/obb/1` in your configuration file. And unfortunately it's a legitimate situation to have a directory with ending of a space:
```
xaionaro@sonic:/tmp$ mkdir -p 'test '
xaionaro@sonic:/tmp$ cd 'test '
xaionaro@sonic:/tmp/test $ 
```
So for string arguments I did not add any trimming and the behaviour remains the same. But I think the ticket can be closed, now.
